indexing_options.go

```diff
@@ -0,0 +1,66 @@
+//  Copyright (c) 2014 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+type FieldIndexingOptions int
+
+const (
+	IndexField FieldIndexingOptions = 1 << iota
+	StoreField
+	IncludeTermVectors
+	DocValues
+)
+
+func (o FieldIndexingOptions) IsIndexed() bool {
+	return o&IndexField != 0
+}
+
+func (o FieldIndexingOptions) IsStored() bool {
+	return o&StoreField != 0
+}
+
+func (o FieldIndexingOptions) IncludeTermVectors() bool {
+	return o&IncludeTermVectors != 0
+}
+
+func (o FieldIndexingOptions) IncludeDocValues() bool {
+	return o&DocValues != 0
+}
+
+func (o FieldIndexingOptions) String() string {
+	rv := ""
+	if o.IsIndexed() {
+		rv += "INDEXED"
+	}
+	if o.IsStored() {
+		if rv != "" {
+			rv += ", "
+		}
+		rv += "STORE"
+	}
+	if o.IncludeTermVectors() {
+		if rv != "" {
+			rv += ", "
+		}
+		rv += "TV"
```

Comment:
I'd rather we used full names here as opposed to acronyms - TERMVECTORS, DOCVALUES.

Reply:
ok, but i'm just moving shit around, if you want to change the functionality we should put that in a separate change and discuss it

Reply:
Oh I see, didn't realize this has always been around.

Reply:
It was document.IndexingOptions in the past.  I renamed it to FieldIndexingOptions here because inside the index package IndexingOptions felt too generic, and no longer sufficiently described it's purpose.